sklearn/cluster/k_means_.py

```diff
@@ -1415,3 +1416,25 @@ def predict(self, X):
 
         X = self._check_test_data(X)
         return self._labels_inertia_minibatch(X)[0]
+   
+    def _reset_estimator(self):
```

Comment:
I don't think this is the fix I would have done.
Where in the ``fit`` method is the old result used?